src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -106,7 +106,7 @@ public TResponse UnaryCall(TRequest msg)
                 using (var metadataArray = MetadataArraySafeHandle.Create(details.Options.Headers))
                 using (var ctx = BatchContextSafeHandle.Create())
                 {
-                    call.StartUnary(ctx, payload, metadataArray, GetWriteFlagsForCall());
+                    call.StartUnary(ctx, payload, GetWriteFlagsForCall(), metadataArray, details.Options.Flags);
```

Comment:
So the API involves creating a CallOptions, used to initialize the call, and attaching a WriteOptions, used for sending messages for the rest of the call? Subjective, but one thing is the `Flags` within CallOptions only including CallOptions-specific flags, but still having the attached `WriteFlags` field with its own flags seemed like it could be a little confusing. Maybe they could be separated more?